uservice_ccutter/plugins/projecttypes/generic.py

```diff
@@ -2,7 +2,34 @@
 """
 import datetime
 
+from apikit import BackendError
+
+
 
 def current_year():
     """Return current year as string."""
     return str(datetime.datetime.now().year)
+  
+
+def raise_ise(text):
+    """Turn error text into a BackendError Internal Server Error.  Handy
+    for reraising exceptions in an easy-to-consume-by-the-client form.
+    """
+    if isinstance(text, Exception):
+        # Just in case we are exuberantly passed the entire Exception and
+        #  not its textual representation.
+        text = str(text)
+    raise BackendError(status_code=500,
+                       reason="Internal Server Error",
+                       content=text)
+
+
+def raise_from_req(req):
```

Comment:
Raise_from_req takes an HTTP response and turns it into an exception if it's 4xx or 5xx.  Requests has a similar method; the idea here is that I want to return a BackendError, whose body contains useful data about the way the call just failed.